my_enumerables.rb

```diff
@@ -0,0 +1,190 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+
+    if pro
+      while i < arr.length
+        proc.cal arr[i]
+        i += 1
+      end
+    else
+      while i < arr.length
+        yield arr[i]
+        i += 1
+      end
+    end
+    arr
+  end
```

Comment:
#my_each when a block is given when self is a range returns the range itself after calling the 
given block once for each element in self